feat: LEAP-1893: LEAP-1900: Add setting to toggle region looping and autoplay

hlomzik opened the pull request:
Two new settings:
- Toggle region looping when playing; default value is changed to false to not annoy users.
- Toggle auto-play of new segments (unlabeled regions); default stays true.

AudioUltra `view.tsx` was changed to move provider with settings higher so they can be used by both Audio and Controls.
Settings are stored in localStorage between sessions.

<table border="0"><tr><td valign="top">
Before<br/>
<img width="220" alt="Screenshot 2025-03-06 at 14 12 57" src="https://github.com/user-attachments/assets/4bac4e01-f05a-4854-a144-1e324ba518ce" />
</td><td>
After<br/>
<img width="272" alt="Screenshot 2025-03-06 at 14 09 21" src="https://github.com/user-attachments/assets/500f8cb6-d20a-4bb1-81be-c331ac9b36a3" />
</td></tr></table>

Improved styles for audio controls:
- uniform font size for all controls
- increase modal width to fit new settings
- adjust paddings for better apearance
- use our common tooltips for info icon

Fixed styles and types for uikit's toggle and label:
- proper ref typing
- fix issues
- remove font-size from a wrong place - it was overriding sizing params

Added `usePersistentState` and `usePersistentJSONState` hooks to core lib. Added tests for them.

hlomzik committed:
feat: LEAP-1893: Add setting to toggle region looping

Default value is changed to false to not annoy users.
In order to do so AudioUltra view was changed to move provider with settings higher.

hlomzik committed:
Fix playing when cursor is outside of selected region

Previously if it was after the region or just at the end of it,
the cursor was moved to the end of the region,
which stops playing if looping is disabled.
Now it's always moved to the beginning of the region.

hlomzik committed:
feat: LEAP-1900: Add setting to disable autoplaying of new audio regions

hlomzik committed:
Improve styles for audio controls

- uniform font size for all controls
- increase modal width to fit new settings
- adjust paddings for better apearance
- use our common tooltips for info icon

hlomzik committed:
Fix styles and types for uikit's toggle and label

- proper ref typing
- fix issues
- remove font-size from a wrong place - it was overriding sizing params

hlomzik committed:
Fix linting

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13725073767)
> Successfully merged:  create mode 100644 web/tsconfig.json

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1893/audio-regions-settings'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13725073767

hlomzik committed:
Add usePersistent[JSON]State hooks; use them for audio tag settings

Hook is moved from LSE to core lib.

hlomzik committed:
Add tests for new hooks; add hooks testing library

hlomzik committed:
Merge branch 'fb-leap-1893/audio-regions-settings' of github.com:HumanSignal/label-studio into fb-leap-1893/audio-regions-settings

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13767497471)
> Successfully merged:  create mode 100644 web/apps/labelstudio/src/utils/query-client.ts

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1893/audio-regions-settings'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13767497471

Gondragos commented on web/libs/editor/src/tags/object/AudioUltra/view.tsx:
```diff
@@ -180,49 +165,80 @@ const AudioUltraView: FC<AudioUltraProps> = ({ item }) => {
-export const AudioUltra = observer(AudioUltraView);
+const AudioUltraWithSettings: FC<AudioUltraProps> = ({ item }) => {
+  const [settings, setSettings] = usePersistentJSONState<TimelineSettings>("ls:audio-tag:settings", {
+    playpauseHotkey: "audio:playpause",
```
Since it is persistent, I am starting to have doubts about whether it is possible to store keyboard shortcuts here.
It looks like it would be hard to change hotkeys in the future.

hlomzik replied:
Yeah, I don't even like the idea of this hotkey to be a setting like this and it's not even possible to change it.

hlomzik replied:
Added a comment about this hotkey, thank you!

Gondragos commented:
Excellent improvements!  And it's great that the hook has been moved to the core library, and now with tests.  This was all sorely lacking. I only have doubts about one setting, which is now also persistent, but perhaps that was unnecessary for it.

hlomzik committed:
Add comment about hotkey in settings

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13782423827)
> Successfully merged:  create mode 100644 web/libs/ui/src/lib/enterprise-badge/enterprise-badge.tsx

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-1893/audio-regions-settings'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13782423827

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13782774847)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13782774847

nikitabelonogov commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13786369470)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13786369470

hlomzik committed:
Use tooltip from ui kit

hlomzik committed:
Fix applying settings at waveform loading

hlomzik committed:
Merge branch 'fb-leap-1893/audio-regions-settings' of github.com:HumanSignal/label-studio into fb-leap-1893/audio-regions-settings

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13788592646)
> Successfully merged: Already up to date.

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13806531709)
> Successfully merged:  delete mode 100644 label_studio/templates/401.html

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-1893/audio-regions-settings'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13806531709

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13806781279)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13806781279

hlomzik committed:
Fix playhead stuck at the end of the new region

It happens in a lot of cases, because playhead could be moved to the end of the region
and in this case `updateLoop()` pauses the audio if looping is disabled.
So we have to explicitly update the current time to the start of the region.

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13833248022)
> Successfully merged:  create mode 100644 label_studio/projects/tests/test_project_sample_task.py

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-1893/audio-regions-settings'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13833248022

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13833406759)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13833406759
